_ml-commons-plugin/tutorials/rag-conversational-agent.md

```diff
@@ -0,0 +1,838 @@
+---
+layout: default
+title: RAG chatbot with a conversational flow agent
+parent: Tutorials
+nav_order: 40
+---
+
+# Retrieval-augmented generation chatbot with a conversational flow agent
+
+This tutorial explains how to use a conversational flow agent to build a retrieval-augmented generation (RAG) application with your OpenSearch data as a knowledge base.
+
+Replace the placeholders starting with the prefix `your_` with your own values.
+{: .note}
+
+An alternative way to build RAG conversational search is to use a RAG pipeline. For more information, see [Conversational search using the Cohere Command model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/tutorials/conversational-search-cohere/).
+
+## Prerequisite
+
+In this tutorial, you'll build a RAG application, which provides an OpenSearch [k-NN index]({{site.url}}{{site.baseurl}}/search-plugins/knn/knn-index/) as a knowledge base for a large language model (LLM). For data retrieval, you'll use [semantic search]({{site.url}}{{site.baseurl}}/search-plugins/semantic-search/). For a comprehensive semantic search tutorial, see [Neural search tutorial]({{site.url}}{{site.baseurl}}/search-plugins/neural-search-tutorial/).
+
+First, you'll need to update cluster settings. If you don't have a dedicated ML node, set `"plugins.ml_commons.only_run_on_ml_node": false`. To avoid triggering a native memory circuit breaker, set `"plugins.ml_commons.native_memory_threshold"` to 100%:
+
+```json
+PUT _cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.only_run_on_ml_node": false,
+        "plugins.ml_commons.native_memory_threshold": 100,
+        "plugins.ml_commons.agent_framework_enabled": true
+    }
+}
+```
+{% include copy-curl.html %}
+
+## Step 1: Prepare the knowledge base
+
+The following steps prepare the knowledge base to supplement an LLM knowledge.
```

Comment:
Should "base" follow "knowledge?"

Reply:
No, this is just the knowledge of the LLM